apps/service_providers/llm_service/runnables.py

```diff
@@ -382,22 +387,56 @@ def _save_response_annotations(self, output, thread_id, run_id) -> tuple[str, di
                             # Original citation text example: sandbox:/mnt/data/the_file.csv.
                             # This is the link part in what looks like
                             # [Download the CSV file](sandbox:/mnt/data/the_file.csv)
-                            session_id = self.state.session.id
                             output_message = output_message.replace(
                                 file_ref_text, f"file:{team.slug}:{session_id}:{created_file.id}"
                             )
-                            generated_files.append(created_file)
-
+                            file_path_attachments.append(created_file)
                         file_ids.add(file_id)
+                else:
+                    # Ignore any other type for now
+                    pass
 
         # Attach the generated files to the chat object as an annotation
-        if generated_files:
-            chat = self.state.session.chat
+        if file_path_attachments:
             resource, _created = chat.attachments.get_or_create(tool_type="file_path")
-            resource.files.add(*generated_files)
+            resource.files.add(*file_path_attachments)
+
+        if image_file_attachments:
+            resource, _created = chat.attachments.get_or_create(tool_type="image_file")
+            resource.files.add(*image_file_attachments)
 
         return output_message, list(file_ids)
 
+    def _create_image_file_from_image_message(self, client, image_file_message) -> File | None:
+        """
+        Creates a File record from `image_file_message` by pulling the data from OpenAI. Typically, these files don't
+        have extentions, so we'll need to guess it based on the content. We know it will be an image, but not which
+        extention to use.
+        """
+        from mimetypes import guess_extension
+
+        import magic
+
+        from apps.assistants.sync import get_and_store_openai_file
+
+        try:
+            file_id = image_file_message.file_id
+            openai_file = client.files.retrieve(file_id=file_id)
+            created_file = get_and_store_openai_file(
+                client=client,
+                file_name=f"{openai_file.filename}",
+                file_id=file_id,
+                team_id=self.state.experiment.team_id,
+            )
+            mimetype = magic.from_buffer(created_file.file.open().read(), mime=True)
+            extention = guess_extension(mimetype)
```

Comment:
create an allowlist for the file types we expect for photos, we don't want any malicious file types getting through

Reply:
Since this is coming from the bot I think it's unlikely to be malicious.

Reply:
These files are generated by the assistant, so I doubt it will be malicious?

Reply:
> Since this is coming from the bot I think it's unlikely to be malicious.

I only saw this after reloading the page :smile: